src/main/kotlin/app/revanced/patches/all/directory/ChangeDataDirectoryLocationPatch.kt

```diff
@@ -0,0 +1,74 @@
+package app.revanced.patches.all.directory
+
+import app.revanced.patcher.extensions.InstructionExtensions.getInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import app.revanced.patches.all.misc.transformation.BaseTransformInstructionsPatch
+import com.android.tools.smali.dexlib2.iface.ClassDef
+import com.android.tools.smali.dexlib2.iface.Method
+import com.android.tools.smali.dexlib2.iface.instruction.Instruction
+import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.ReferenceInstruction
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+import com.android.tools.smali.dexlib2.iface.instruction.formats.Instruction35c
+import com.android.tools.smali.dexlib2.immutable.reference.ImmutableMethodReference
+import com.android.tools.smali.dexlib2.util.MethodUtil
+
+@Patch(
+    name = "Change data directory location",
+    description = "Changes data directory in the application from data/0 to android/data for root-less devices.",
```

Comment:
Please mention the full path here:

```suggestion
    description = "Changes the data directory in the application from data/0 to android/data for root-less devices.",
```